enumerables.rb

```diff
@@ -0,0 +1,165 @@
+        to_change << arr[i] if yield(arr[i])
+        i += 1
+      end
+      to_change
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_any?(arg = nil)
+    to_change = false
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = true unless obj == false || obj.nil? }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = true if obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = true if obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = true if obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = true if yield obj }
+    end
+    to_change
+  end
+
+  def my_all?(arg = nil)
+    to_change = true
+    arr = self
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = false unless arr.include?(true) || obj != false }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = false unless obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = false unless obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = false unless obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = false unless yield obj }
+    end
+    to_change
+  end
+
+  def my_none?(arg = nil)
+    to_change = true
+    arr = self
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = false if arr.include?(true) || obj == true }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = false if obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = false if obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = false if obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = false if yield obj }
+    end
+    to_change
+  end
+
+  def my_map
+    arr = self
+    return arr.to_enum unless block_given?
+
+    to_change = []
+    my_each do |obj|
+      to_change.push(yield obj)
+    end
+    to_change
+  end
+
+  def my_count(arg = nil)
+    arr = self
+    total = 0
+    if arg.nil?
+      if block_given?
+        my_each do |item|
+          total += 1 if yield(item)
+        end
+      else
+        total = arr.length
+      end
+    else
+      my_each { |obj| total += 1 if obj == arg }
+    end
+    total
+  end
+
+  def my_inject(start_value = 0, operator = nil)
+    arr = self
+    to_change = 0
+    to_change = 1 if operator == :*
+    to_change = 1 if operator == :/
+    to_change = start_value if start_value != 0
+    operator = start_value if start_value.is_a? Symbol
+    if operator.is_a? Symbol
+      case operator
+      when :+
+        arr.each { |obj| to_change += obj }
+      when :-
+        arr.each { |obj| to_change -= obj }
+      when :*
+        arr.each { |obj| to_change *= obj }
+      when :/
+        arr.each { |obj| to_change /= obj }
+      end
```

Comment:
Don't you trust your own my_each method? If you are confused, I am assuring you, your my_each works totally fine. Now there is a better way to handle this symbol. You can use the send method. Check out this [example](https://stackoverflow.com/questions/3337285/what-does-send-do-in-ruby).